File: Decorator/DecoratorCoffee/DecoratorCoffee/SugarDecorator.cs
namespace DecoratorCoffee
{
    public class SugarDecorator : BaseDecorator
    {
        public SugarDecorator(ICoffee coffee) : base(coffee)
        {
        }

        public override double GetCost() => base.GetCost() + 0.25;
        public override string GetDescription() => base.GetDescription() + ", sugar";    
    }
}

File: Decorator/Java/DecoratorCoffee/src/SugarDecorator.java
public class SugarDecorator extends BaseDecorator{
    public SugarDecorator(ICoffee coffee) {
        super(coffee);
    }

    @Override
    public double getCost() {
        return super.getCost() + 0.25;
    }

    @Override
    public String getDescription() {
        return super.getDescription() + ", sugar";
    }
}
